provider/datagen/src/transform/cldr/source.rs

```diff
@@ -9,71 +9,51 @@ use std::fmt::Debug;
 use std::str::FromStr;
 
 /// Specifies a variant of CLDR JSON
-#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, serde::Deserialize)]
 #[non_exhaustive]
-pub enum LocaleSubset {
-    /// Includes all data
-    Full,
-    /// Includes locales listed as modern coverage targets by the CLDR subcomittee
+pub enum CoverageLevel {
+    /// Locales listed as modern coverage targets by the CLDR subcomittee.
+    #[serde(rename = "modern")]
     Modern,
-}
-
-impl std::fmt::Display for LocaleSubset {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(
-            f,
-            "{}",
-            match self {
-                Self::Full => "full",
-                Self::Modern => "modern",
-            }
-        )
-    }
+    /// Locales listed as moderate coverage targets by the CLDR subcomittee.
+    #[serde(rename = "moderate")]
+    Moderate,
+    /// Locales listed as basic coverage targets by the CLDR subcomittee.
+    #[serde(rename = "basic")]
+    Basic,
 }
 
 #[derive(Debug)]
-pub(crate) struct CldrCache {
-    pub locale_subset: LocaleSubset,
-    pub cache: SerdeCache,
-}
+pub(crate) struct CldrCache(pub SerdeCache);
 
 impl CldrCache {
     pub fn core(&self) -> CldrDirNoLang<'_> {
-        CldrDirNoLang(&self.cache, "cldr-core".to_string())
+        CldrDirNoLang(&self.0, "cldr-core".to_owned())
     }
 
     pub fn numbers(&self) -> CldrDirLang<'_> {
-        CldrDirLang(
-            &self.cache,
-            format!("cldr-numbers-{}/main", self.locale_subset),
-        )
+        CldrDirLang(&self.0, "cldr-numbers-full/main".to_owned())
     }
 
     pub fn misc(&self) -> CldrDirLang<'_> {
-        CldrDirLang(
-            &self.cache,
-            format!("cldr-misc-{}/main", self.locale_subset),
-        )
+        CldrDirLang(&self.0, "cldr-misc-full/main".to_owned())
     }
 
     pub fn bcp47(&self) -> CldrDirNoLang<'_> {
-        CldrDirNoLang(&self.cache, "cldr-bcp47/bcp47".to_string())
+        CldrDirNoLang(&self.0, "cldr-bcp47/bcp47".to_string())
     }
 
     pub fn displaynames(&self) -> CldrDirLang<'_> {
-        CldrDirLang(
-            &self.cache,
-            format!("cldr-localenames-{}/main", self.locale_subset),
-        )
+        CldrDirLang(&self.0, "cldr-localenames-full/main".to_owned())
```

Comment:
Actually, this is an Issue: If someone downloads the data themselves (not using our auto-downloader), it's totally reasonable to think that they have the modern package, and we should at the very least support that, since we are a client of CLDR JSON.

Reply:
Can we not just tell them to download the modern package?

Reply:
@srl295 Thoughts on this? Is it useful to keep the code path that lets users read from "modern" CLDR JSON rather than "full" CLDR JSON (due to file name differences)?

Reply:
I would think some users might want to have a smaller subset and just use modern. Especially since the data is fully expanded and thus takes more space than it otherwise would.

Reply:
> Especially since the data is fully expanded and thus takes more space than it otherwise would.

It's not, we never unzip the file and the size difference is not that significant.

With this PR we do support generating only modern locales from the full file, so we don't really *need* to support the modern file.

Reply:
I think Steven meant "fully expanded" in the sense that all inheritance labels are resolved, so the data is larger than the source XML data.

Reply:
If you can add back `--cldr-locale-subset` in a half hour, I prefer to do it. It should only affect two things: the file names for reading from the CLDR directory, and the file that gets downloaded. However, the setting should default to "full", and we leave the setting at its default value in almost all situations.

Reply:
I wouldn't want to add that back in. We should detect what file we have if we want `-modern` support

Reply:
@srl295 why do the subdirectories contain `-modern` and `-full` anyway? Is that something we can change?

Reply:
@robertbastian they are separate packages, so that users who only want the smaller set can request them. See https://github.com/unicode-org/cldr-json/blob/main/PACKAGES.md

Reply:
But the packaged releases don't *have to* use those package names. But I guess that ship has sailed to stability land.